src/internal/connector/exchange/container_resolver.go

```diff
@@ -40,50 +48,147 @@ type containersEnumerator interface {
 // folders if each folder is only a single character.
 const maxIterations = 300
 
-func newContainerResolver() *containerResolver {
+func newContainerResolver(refresher containerRefresher) *containerResolver {
 	return &containerResolver{
-		cache: map[string]graph.CachedContainer{},
+		cache:     map[string]graph.CachedContainer{},
+		refresher: refresher,
 	}
 }
 
 type containerResolver struct {
-	cache map[string]graph.CachedContainer
+	cache     map[string]graph.CachedContainer
+	refresher containerRefresher
 }
 
 func (cr *containerResolver) IDToPath(
 	ctx context.Context,
 	folderID string,
 ) (*path.Builder, *path.Builder, error) {
-	return cr.idToPath(ctx, folderID, 0)
+	ctx = clues.Add(ctx, "container_id", folderID)
+
+	c, ok := cr.cache[folderID]
+	if !ok {
+		return nil, nil, clues.New("folder not cached").WithClues(ctx)
+	}
+
+	p := c.Path()
+	if p == nil {
+		return nil, nil, clues.New("folder has no path").WithClues(ctx)
```

Comment:
"cached folder" removes ambiguity about whether it's the msoft data or cache data missing a path.
```suggestion
		return nil, nil, clues.New("cached folder has no path").WithClues(ctx)
```